backend/internal/service/handler/users/follow.go

```diff
@@ -0,0 +1,72 @@
+package users
+
+import (
+	"github.com/gofiber/fiber/v2"
+	"github.com/google/uuid"
+)
+
+type FollowRequest struct {
+	FollowerId  uuid.UUID `json:"followerId"`
+	FollowingId uuid.UUID `json:"followingId"`
+}
+
+func (h *Handler) FollowUnfollowUser(c *fiber.Ctx) error {
+	var body FollowRequest
+
+	if err := c.BodyParser(&body); err != nil {
+		return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{
+			"error": "Cannot parse JSON",
+		})
+	}
+
+	followerStr := body.FollowerId.String()
+	followingStr := body.FollowingId.String()
+
+	// Get the current user ID (inferred from session, authentication middleware)
+	// currentUserId := c.Locals("userId").(string)
+	follower, err := h.userRepository.UserExists(c.Context(), followerStr)
+	if err != nil || !follower {
+		return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{
+			"error": "User Not found",
+		})
```

Comment:
you can use 		
`return errs.NotFound("Follower", "id", followerStr)`
which is a custom function we built to return an error formatted with the id!